runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -652,7 +660,18 @@ function LanguageTree:_get_injection(match, metadata)
 
     -- Lang should override any other language tag
     if name == 'injection.language' then
-      lang = vim.treesitter.get_node_text(node, self._source, { metadata = metadata[id] })
+      local text =
+        vim.treesitter.get_node_text(node, self._source, { metadata = metadata[id] }):lower()
```

Comment:
```suggestion
        vim.treesitter.get_node_text(node, self._source, { metadata = metadata[id] })
```

As mentioned, I'd much prefer that we didn't do this.

Reply:
I still haven't heard a compelling argument that outweighs the significant benefits (no need for custom directives that make it harder to share queries).

Reply:
The benefits aren't significant IMO, and it implies that treesitter languages must be case-insensitive which isn't something we can decide or enforce.

Unless upstream makes this rule, we shouldn't do this.

Reply:
Well, then it stands opinion against opinion.

Reply:
As a compromise we could try `lower()` as a fallback if nothing else resolves. That way we don't force case insensitivity.